gateway/redis_signals.go

```diff
@@ -322,3 +325,55 @@ func (gw *Gateway) handleDashboardZeroConfMessage(payload string) {
 		pubSubLog.Info("Hostname set with dashboard zeroconf signal")
 	}
 }
+
+// handleUserKeyReset processes a user key reset notification
+func (gw *Gateway) handleUserKeyReset(payload string) {
+	keys := strings.Split(payload, ":")
+	if len(keys) != 2 {
+		log.Error("Invalid user key reset payload")
+		return
+	}
+
+	keys = strings.Split(keys[0], ".")
+	if len(keys) != 2 {
+		log.Error("Invalid user key reset payload")
+		return
+	}
+
+	oldKey := keys[0]
+	newKey := keys[1]
+
+	config := gw.GetConfig()
+
+	if oldKey == config.SlaveOptions.APIKey {
+		config.SlaveOptions.APIKey = newKey
+		gw.SetConfig(config)
+
+		// If we're using a KV store, update the API key there as well
+		if config.SlaveOptions.OriginalAPIKeyPath != "" {
```

Comment:
here you could assign `config.SlaveOptions.OriginalAPIKeyPath` to a var so it will look clearner below:

```
if keyPath!= ""{
    if strings.HasPrefix("vault://", keyPath)
}
.
.
.
```